pkg/hive/example/mini/main.go

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+// A minimal example implementation of dependency injection in Go using reflection.
+// For educational purposes only :-)
+
+package main
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type A struct{}
+
+func NewA() *A {
+	return &A{}
+}
+
+type B struct {
+	A *A
+}
+
+func NewB(A *A) *B {
+	return &B{A}
+}
+
+func showB(B *B) {
+	fmt.Printf("B: %#v\n", B)
+}
+
+func main() {
+	c := container{
+		providers: make(map[int]provider),
+		byType:    make(map[string]int),
+		objects:   make(map[string]reflect.Value),
+	}
+
+	c.provide(NewA)
+	c.provide(NewB)
+
+	c.invoke(showB)
+}
+
+type provider struct {
+	ctor any
+	ins  []reflect.Type
+	out  []reflect.Type
+}
+
+type container struct {
+	nextId    int
+	providers map[int]provider
+	byType    map[string]int
+	objects   map[string]reflect.Value
+}
+
+func (c *container) provide(ctor any) {
+	typ := reflect.TypeOf(ctor)
+	in := make([]reflect.Type, typ.NumIn())
+	for i := 0; i < typ.NumIn(); i++ {
+		in = append(in, typ.In(i))
+	}
```

Comment:
I think this should either be

```suggestion
	in := make([]reflect.Type, 0, typ.NumIn())
	for i := 0; i < typ.NumIn(); i++ {
		in = append(in, typ.In(i))
	}
```

or

```suggestion
	in := make([]reflect.Type, typ.NumIn())
	for i := 0; i < typ.NumIn(); i++ {
		in[i] = typ.In(i)
	}
```

In its current form `in` will contain `typ.NumIn()` zero values followed by `typ.NumIn()` input parameters which is probably not intended.

Reply:
Oops heh, of course the quick-and-dirty-never-compiled example was wrong. Will fix, thanks!